test/e2e/operator_test.go

```diff
@@ -228,3 +249,199 @@ func setImage(deployment *appsv1.Deployment, image string) {
 		}
 	}
 }
+
+func TestDNSForwarding(t *testing.T) {
+	cl, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create the upstream resolver ConfigMap.
+	upstreamCfgMap := buildConfigMap(upstreamPodName, upstreamPodNs, "Corefile", upstreamCorefile)
+	if err := cl.Create(context.TODO(), upstreamCfgMap); err != nil {
+		t.Fatalf("failed to create configmap %s/%s: %v", upstreamCfgMap.Namespace, upstreamCfgMap.Name, err)
+	}
+	defer func() {
+		if err := cl.Delete(context.TODO(), upstreamCfgMap); err != nil {
+			t.Fatalf("failed to delete configmap %s/%s: %v", upstreamCfgMap.Namespace, upstreamCfgMap.Name, err)
+		}
+	}()
+
+	// Get the CoreDNS image used by the test upstream resolver.
+	co := &configv1.ClusterOperator{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: operatorcontroller.DNSOperatorName}, co); err != nil {
+		t.Fatalf("failed to get clusteroperator %s: %v", operatorcontroller.DNSOperatorName, err)
+	}
+	var (
+		coreImage      string
+		coreImageFound bool
+	)
+	for _, ver := range co.Status.Versions {
+		if ver.Name == operatorcontroller.CoreDNSVersionName {
+			if len(ver.Version) == 0 {
+				t.Fatalf("clusteroperator %s has empty coredns version", operatorcontroller.DNSOperatorName)
+				break
```

Comment:
You can drop the `t.Fatalf` (or drop the `break` if you prefer).

Reply:
ack